File: PROFESORES/MDA/KAFKA/src/main/java/kafka_tutorial/exercise_02_basic_producer_consumer/Consumer.java
package kafka_tutorial.exercise_02_basic_producer_consumer;

import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.clients.consumer.ConsumerRecords;
import org.apache.kafka.clients.consumer.KafkaConsumer;

import java.time.Duration;
import java.util.Arrays;

import java.util.Properties;
import java.util.UUID;

public class Consumer {

    public static void main(String[] args) {
        Properties properties = new Properties();
        properties.put("bootstrap.servers", "localhost:9092");
        properties.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        properties.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        properties.put("auto.offset.reset", "earliest");
        properties.put("group.id", "AppIdentifer005");

        KafkaConsumer<String, String> kafkaConsumer = new KafkaConsumer<String, String>(properties);

        //Subscribe
        kafkaConsumer.subscribe(Arrays.asList("myTopic"));

        try{
            while (true){
                ConsumerRecords<String, String> records = kafkaConsumer.poll(Duration.ofMillis(100));
                for (ConsumerRecord<String, String> record: records){
                    System.out.println(String.format("Topic - %s, Partition - %d, Value: %s", record.topic(), record.partition(), record.value()));
                }
            }
        }catch (Exception e){
            System.out.println(e.getMessage());
        }finally {
            kafkaConsumer.close();
        }
    }
}

File: ALUMNOS/MDAB/FABIAN_MIULESCU/KAFKA/consumer.py
from confluent_kafka import Consumer, KafkaError
import json

config = {
    'bootstrap.servers': 'localhost:9092',
    'group.id': 'retail-consumer-group',
    'auto.offset.reset': 'earliest'
}

consumer = Consumer(config)
topic = 'retail'
consumer.subscribe([topic])

try:
    print("▶️  Listening to topic 'retail'...")
    while True:
        msg = consumer.poll(1.0)
        if msg is None:
            continue
        if msg.error():
            if msg.error().code() == KafkaError._PARTITION_EOF:
                continue
            else:
                print("Error receiving message:", msg.error())
        else:
            data = json.loads(msg.value().decode('utf-8'))
            print(f" New message: {data}")
except KeyboardInterrupt:
    pass
finally:
    consumer.close()
    print("🛑 Consumer closed.")
